### ISSUE TITLE
Ratelimit issues

### ISSUE THREAD

**Issue by atomfrede** [ethomson/issue-dashboard#2]
Hi @ethomson 

Thanks for the nice workflow. I have setup a small [dashboard for the jhipster org on github](https://hipster-labs.github.io/github-dashboard/). But we are running now in rate limiting issues (the schedule was at first every hour). Maybe it is related to the configuration as I would like to display more projects than currently on the generated dashboard and thats why running into rate limit.

Is there a way to pass an api user or so, such that the the workflow runs on the higher limit? 

If it helps the configuration is here: https://github.com/hipster-labs/github-dashboard

Thanks for help or any hint in advance!

**Comment by ethomson** [comment]
Neat!

Sorry to hear about the rate limit issues.  Right now your configuration is running with the github token that's generated for your workflow run.  My recollection is that this has the same API limits as a normal user - but I could be wrong.  You could try generating a PAT and using that as your token instead (https://github.com/hipster-labs/github-dashboard/blob/master/.github/workflows/dashboard.yml#L272).

But ultimately I think that this action just needs to be smarter about the way it collects data.  GitHub hands us back the rate limiting information (how many additional calls you can make, when your limit refreshes), we could take advantage of that to pause data collection until there's a refresh instead of failing miserably like we do now.

**Comment by atomfrede** [comment]
Thanks for the quick reply. Will try a personal access token.

**Comment by ethomson** [comment]
Yeah - reading [this page](https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions) suggests that you'll only get 1000 API requests with the github token and you should get 5x more with a PAT.  Let me know if that's not enough and we should be able to think about how to pause to keep the rate limit in check.

**Comment by atomfrede** [comment]
:thinking: It seems not to work (which is strange when looking at the limits). My personal access token has `public_repo` scope. I scheduled the job for noon maybe something is wrong.

**Comment by atomfrede** [comment]
Still the same. Rate limit exceeded for my user, even with two projects less.
